apps/voting/contracts/Voting.sol

```diff
@@ -453,16 +676,64 @@ contract Voting is IForwarder, AragonApp {
     }
 
     /**
-    * @dev Internal function to check if a voter can participate on a vote. It assumes the queried vote exists.
+    * @dev Internal function to check if the vote is open and given option is applicable at the current phase.
+    *      It assumes the queried vote exists.
+    * @param vote_ The queried vote
+    * @param _supports Whether the voter supports the vote or not
     * @return True if the given voter can participate a certain vote, false otherwise
     */
-    function _canVote(uint256 _voteId, address _voter) internal view returns (bool) {
-        Vote storage vote_ = votes[_voteId];
-        return _isVoteOpen(vote_) && token.balanceOfAt(_voter, vote_.snapshotBlock) > 0;
+    function _isValidPhaseToVote(Vote storage vote_, bool _supports) internal view returns (bool) {
+        return _isVoteOpen(vote_) && (!_supports || _getVotePhase(vote_) == VotePhase.Main);
+    }
+
+    /**
+    * @dev Internal function to check if the _delegate is a current delegate for the _voter
+    * @param _delegate address of the delegate
+    * @param _voter address of the voter
+    * @return True if _delegate is a current delegate for the _voter, false otherwise
+    */
+    function _isDelegateFor(address _delegate, address _voter) internal view returns (bool) {
```

Comment:
I would suggest to rename `_isDelegateForVoter` or `_isDelegated`. Just because typical function named `somethingFor` implies that first arg is something for what this function is being applyed